### PR TITLE
(#6011) Refactor AX parser to use electricitymaps event classes

### PR DESCRIPTION
**Pull request by AlisCode**
## Issue

#6011

## Description

This PR upgrades the parser for Aaland (ZoneKey: AX) to use electricitymaps' event classes, as described in the issue.

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: parsers/AX.py [hunk 1]
```diff
@@ -84,48 +92,40 @@ def formated_data(
     )
     if date > date_time:
         date = date - timedelta(days=1)
-    return_list = []
+    exchanges = ExchangeList(logger)
+    production_breakdowns = ProductionBreakdownList(logger)
+    consumption = TotalConsumptionList(logger)
     for data in data_list:
         corrected_date = date - timedelta(minutes=15 * data_list.index(data))
         if data_type == "production":
```
Comment (VIKTORVAV99) [inline parsers/AX.py:100]:
```suggestion
        if data_type == "production" and zone_key is not None:
```

File: parsers/AX.py [hunk 2]
```diff
@@ -84,48 +92,40 @@ def formated_data(
...
-                    "source": SOURCE,
-                }
+            production_mix = ProductionMix(wind=data["wind"], oil=data["fossil"])
+            production_breakdowns.append(
+                datetime=corrected_date,
+                production=production_mix,
+                source=SOURCE,
+                zoneKey=zone_key,
             )
         elif data_type == "consumption":
```
Comment (VIKTORVAV99) [inline parsers/AX.py:108]:
```suggestion
        elif data_type == "consumption" and zone_key is not None:
```

**Review by VIKTORVAV99 — COMMENTED** [review]
Thanks for doing this!
The changes look good functionality wise but I left suggestions to remove two type errors.

**Comment by AlisCode** [comment]
Hi @VIKTORVAV99 , thank you for reviewing :)

I fixed the type errors as you suggest (even though I think it would have been best to just make the argument `zone_key` of `formatted_data` not `None` ? I looked at the call sites and `zone_key` is always defined). 

Out of curiosity for next time I want to contribute, what are you guys' tooling to typecheck Python stuff ?

**Comment by VIKTORVAV99** [comment]
> Hi @VIKTORVAV99 , thank you for reviewing :)
> 
> 
> 
> I fixed the type errors as you suggested (even though I think it would have been best to just make the argument `zone_key` of `formatted_data` a `ZoneKey` as opposed to `ZoneKey | None` ? I looked at the call sites and `zone_key` is always defined). 
> 
> 
> 
> Out of curiosity for next time I want to contribute, what are you guys' tooling to typecheck Python stuff ?

It would not be defined for the exchange data where it's assigned a none value.

In the CI we use RUFF to run static analysis and locally I use the pylance plugin for VSCode that implements the pyright type checker. We plan on running this in the CI as well when we have removed the existing type errors.

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM!

Thanks for updating it!

**Comment by AlisCode** [comment]
> It would not be defined for the exchange data where it's assigned a none value.

Well that one slipped past, and is indeed quite obvious. That's why you need tooling :) 

Thank you for the merge and the tooling recommendation!

**Pull request merged**